src/windshaft/named-map.js

```diff
@@ -13,13 +14,18 @@ var NamedMap = MapBase.extend({
     //       "2": "/** torque visualization */\n\nMap { ... }"
     //     }
     //   }
-    //
-    json.styles = this._layersCollection.reduce(function (p, c, i) {
-      var style = c.get('cartocss');
+
+    var layers = this._layersCollection.filter(function (layer) {
+      return layer.get('type') !== 'GMapsBase';
```

Comment:
Mmm I don't think so, `getCartoDBLayers` will return only `CARTODB_LAYER_TYPE` layers. But the thing is, if the map doesn't have a google basemap, it should return the tile layer, otherwise the indexes will change for this case too, and the autostyle will break for embeds not using google basemap.